client/question/services/EventHandlerService.js

```diff
@@ -217,6 +217,22 @@ tie.factory('EventHandlerService', [
         }
       },
 
+      /**
+       * Submits data to TIE's backend to create a CodeRestoreEvent.
+       *
+       */
+      createCodeRestoreEvent: function() {
+        if (ServerHandlerService.doesServerExist()) {
+          _currentEventBatch.push({
+            type: 'CodeRestoreEvent',
+            data: {
+              createdMsec: (new Date()).getTime()
+            }
+          });
+          sendCurrentEventBatch();
```

Comment:
My bad for not mentioning this, but I don't think a CodeRestoreEvent should count as a terminal event -- it shouldn't prompt us sending the event batch. You can just drop this line.